RockWeb/Blocks/Finance/TransactionEntry.ascx.cs

```diff
@@ -1124,33 +1185,96 @@ private void GetAccounts()
             SelectedAccounts = new List<AccountItem>();
             AvailableAccounts = new List<AccountItem>();
 
-            // Enumerate through all active accounts that are public
-            foreach ( var account in new FinancialAccountService( rockContext ).Queryable()
+            // Limit selections to accounts passed through URL
+            if ( _parameterAccounts.Count > 0 )
+            {
+                foreach ( var acct in _parameterAccounts )
+                {
+                    var accountItem = new AccountItem( acct.Id, acct.Order, acct.Name, acct.CampusId, acct.PublicName );
+                    SelectedAccounts.Add( accountItem );
+                }
+            }
+            else
+            {
+                // Enumerate through all active accounts that are public
+                foreach ( var account in new FinancialAccountService( rockContext ).Queryable()
                 .Where( f =>
                     f.IsActive &&
                     f.IsPublic.HasValue &&
                     f.IsPublic.Value &&
-                    ( f.StartDate == null || f.StartDate <= RockDateTime.Today ) &&
-                    ( f.EndDate == null || f.EndDate >= RockDateTime.Today ) )
+                    (f.StartDate == null || f.StartDate <= RockDateTime.Today) &&
+                    (f.EndDate == null || f.EndDate >= RockDateTime.Today) )
                 .OrderBy( f => f.Order ) )
-            {
-                var accountItem = new AccountItem( account.Id, account.Order, account.Name, account.CampusId, account.PublicName );
-
-                if ( showAll )
-                {
-                    SelectedAccounts.Add( accountItem );
-                }
-                else
                 {
-                    if ( selectedGuids.Contains( account.Guid ) )
+                    var accountItem = new AccountItem( account.Id, account.Order, account.Name, account.CampusId, account.PublicName );
+
+                    if ( showAll )
                     {
-                        SelectedAccounts.Add( accountItem );
+                        if ( _accountCampusContext == -1 ) // no filter
```

Comment:
instead of copying and pasting the same logic (lines 1213-1230, 1236-1253, and 1259-1277), you could do the CampusContext filtering in one swoop after the Accounts are added. In other words, add all the accounts regardless of teh CampusContext, then after that, remove the ones that aren't associated with the CampusContext. Make sense?